### DISCUSSION TITLE
Support for the MCreator community.

### DISCUSSION THREAD

**Discussion by northwesttreesgaming** [bernie-g/geckolib#144] [General]
Hello, I am Kieran Martin the YouTube creator for NorthWestTrees Gaming. I work with the developer of MCreator to create tutorials for their software. I also work with many of the contributors providing testing and suggestions for future support for the feature, which is why I am here today.

In the past, many mod creators in our community used GeckoLib to animate their entities using Blockbench, recently though due to the animation file becoming a separate file this has prevented our contributors and developers of MCreator from implementing support the MCreator code is complex and hard to work with at times so sadly it's not possible to work around this change for the recent update due to limitations in our code.

By any chance would the GeckoLib team be willing to help with adding support for an option for single animations in BlockBench. Our contributors may be able to assist you with the knowledge of MCreator.

**Comment by northwesttreesgaming** [comment]
@AzureDoom
I am taking your thumbs down emoji would be a no then?

I am aware of the tensions between the forge community and MCreator community from working with our contributors. It's been an uphill battle trying to figure out on our own what issues the forge community has issues with many members of the community will not work with us to resolve the issues they have which leave us to assume potential issues that may be the issue.

Our intentions were never to cause conflict with the community and it's been hard to figure out what issues the forge community has because many people are quick to spread false information about the community and not come to us about any issue they may have with the community.

Maybe instead you would be open to discussing some of the issues with us so we can work towards lowering some of the tensions because so far no one has really been willing to work with us to get past the issues they have and we have always been open to working with the forge community to resolve their concerns with MCreator.

Hopefully, we will be able to work something out for the long term, we just need someone willing to work with us to move past this point.

**Reply by AzureDoom** [reply]
I will not be supporting any MCreator user with Geckolib as:
1. We already have enough issues with people being able to install the library normally with their lack of Gradle knowledge. 
2. With the release of 1.17, I do not have the time to deal with MCreator users as I will be busy updating this and my own mods.
3. I do not agree with the use of MCreator personally.

I will not be supporting the use of Geckolib with MCreator at all and this will not be changing for me personally at all. Gecko is aware of this as I had stated these when he first showed me this post. As the current maintainer for the library, this request from me is
![image](https://user-images.githubusercontent.com/7415711/121181593-2e8e5f80-c830-11eb-8e06-80fb91cedaf1.png)

**Reply by bernie-g** [reply]
I support Azure's decision, but for different reasons.

On multiple occasions I've had MCreator users join my discord, ask for help, get impatient, and then start spamming and tagging admins. When we mute them they join on alts and brigade the server. I understand that this is because the majority of MCreator users are children, but it's still very annoying to deal with. I don't want to have any association with that and nobody on the team has the time or patience to help children figure out basic computer skills. Plus, maybe it'll help convince some of them to actually learn Java :P

**Comment by northwesttreesgaming** [comment]
> We already have enough issues with people being able to install the library normally with their lack of Gradle knowledge.

Looking at it as an issue is never a great thing to do, many people are eager to learn how it works so it should be seen as a positive, normally documentation and if you have someone that can do YouTube tutorials this can help a lot for any community learns how things work.

> With the release of 1.17, I do not have the time to deal with MCreator users as I will be busy updating this and my own mods.

Fair enough, I can relate to that as I have videos I work on and a couple of mods that I maintain so I know how priorities can impact a small team like this.

> I do not agree with the use of MCreator personally.

This does not really explain too much whether it's the people in the community or the project itself. Every community will have its bad actors, however many people from what I have heard from some members of the MCreator community have talked to people in the forge community, and like the MCreator community not all of them are bad people. 

I don't disagree that it would be annoying for people to come on to a discord server or forum and start causing issues, however, I just want to point out that we don't agree with them doing this to other communities. This is why it's important to not judge the actions of a few individuals on community values.

I will respect your decision to not work on fixing the issue, I do thank you for your time to discuss this.
If you would like to discuss anything further my business email is northwesttreesgaming@gmail.com
